common/protob/messages-stellar.proto

```diff
@@ -175,10 +199,10 @@ message StellarChangeTrustOp {
  */
 message StellarAllowTrustOp {
     optional string source_account = 1;     // (optional) source account address
-    optional string trusted_account = 2;    // The account being allowed to hold the asset
-    optional uint32 asset_type = 3;         // 1 = 4-character, 2 = 12-character
+    required string trusted_account = 2;    // The account being allowed to hold the asset
+    required StellarAssetType asset_type = 3;
     optional string asset_code = 4;         // human-readable asset code
-    optional uint32 is_authorized = 5;
+    required bool is_authorized = 5;
```

Comment:
I don’t think it should be a bool value here.

https://github.com/stellar/stellar-core/blob/master/src/xdr/Stellar-transaction.x#L269

Reply:
good catch, but it's treated in code as bool anyway:
https://github.com/trezor/trezor-firmware/blob/master/core/src/apps/stellar/operations/serialize.py#L30
https://github.com/trezor/trezor-firmware/blob/master/legacy/firmware/stellar.c#L989

i'm wondering why it isn't defined as an enum in the XDR?

Reply:
It was indeed bool before, but it has changed with the update of the protocol. https://github.com/stellar/stellar-core/commit/29669910a7ef57db6b21dc86ee207fda3cbc1c39#diff-11f6a6577ee4d25e4eb1b1834bd71486cc73c26ad4eda68345286a27f5d74307R236

> i'm wondering why it isn't defined as an enum in the XDR?

I don't know why, but my personal guess is to support bitwise operators?

Reply:
> good catch, but it's treated in code as bool anyway:
> https://github.com/trezor/trezor-firmware/blob/master/core/src/apps/stellar/operations/serialize.py#L30
> https://github.com/trezor/trezor-firmware/blob/master/legacy/firmware/stellar.c#L989

Do you plan to modify it in this PR? At present, if passing `is_authorized` is not 0 or 1, the user cannot get the correct signature.

Reply:
> I don't know why, but my personal guess is to support bitwise operators?

originally yes, apparently, but the comment is pretty explicit and there's even a [review comment](https://github.com/stellar/stellar-core/pull/2881#discussion_r560476438) about it

i'll probably do an enum in our protobuf anyway, but it would be nicer to be in line with upstream about it

Reply:
As mentioned in your PR, I won't be changing this for an operation that is deprecated anyway. We'll leave that to if/when we implement the additional Stellar operations.